stbt_auto_selftest.py

```diff
@@ -147,6 +147,20 @@ def init_worker():
     signal.signal(signal.SIGINT, signal.SIG_IGN)
 
 
+def iterate_with_progress(iterator, width=20, stream=sys.stderr):
+    stream.write('\n')
+    total = len(iterator)
+    for n, v in enumerate(iterator):
+        if n == total:
+            break
+        progress = (n * width) // total
+        stream.write(
+            '\033[K[%s] %8d / %d - Processing %s\r' % (
+                '#' * progress + ' ' * (width - progress), n, total, str(v)))
+        yield v
+    stream.write('\r' + ' ' * (total + 28) + '\r')
```

Comment:
Is this supposed to erase the progressbar after we're done? If so:

* The `28` won't overwrite the `str(v)` at the end of the line.
* Shouldn't it be `' ' * width` rather than `' ' * total`?
* Perhaps it would be easier to leave the progressbar up and write a '\n' at the end.